custom_components/zwave_mqtt/climate.py

```diff
@@ -40,23 +41,47 @@
 
 ATTR_FAN_ACTION = "fan_action"
 
+
+class ThermostatMode(IntEnum):
+    """Enum with all (known/used) Z-Wave ThermostatModes."""
+
+    # https://github.com/OpenZWave/open-zwave/blob/master/cpp/src/command_classes/ThermostatMode.cpp
+    OFF = 0
+    HEAT = 1
+    COOl = 2
+    AUTO = 3
+    AUXILIARY = 4
+    RESUME_ON = 5
+    FAN = 6
+    FURNANCE = 7
+    DRY = 8
+    MOIST = 9
+    AUTO_CHANGE_OVER = 10
+    HEATING_ECON = 11
+    COOLING_ECON = 12
+    AWAY = 13
+    FULL_POWER = 14
+
+
 MODE_SETPOINT_MAPPINGS = {
-    "off": (),
-    "heat": ("setpoint_heating",),
-    "cool": ("setpoint_cooling",),
-    "auto": ("setpoint_heating", "setpoint_cooling"),
-    "aux heat": ("setpoint_heating",),
-    "furnace": ("setpoint_furnace",),
-    "dry air": ("setpoint_dry_air",),
-    "moist air": ("setpoint_moist_air",),
-    "auto changeover": ("setpoint_auto_changeover",),
-    "heat econ": ("setpoint_eco_heating",),
-    "cool econ": ("setpoint_eco_cooling",),
-    "away": ("setpoint_away_heating", "setpoint_away_cooling"),
-    "full power": ("setpoint_full_power",),
+    ThermostatMode.OFF: (),
+    ThermostatMode.HEAT: ("setpoint_heating",),
+    ThermostatMode.COOl: ("setpoint_cooling",),
+    ThermostatMode.AUTO: ("setpoint_heating", "setpoint_cooling"),
+    ThermostatMode.AUXILIARY: ("setpoint_heating",),
+    ThermostatMode.FURNANCE: ("setpoint_furnace",),
+    ThermostatMode.DRY: ("setpoint_dry_air",),
+    ThermostatMode.MOIST: ("setpoint_moist_air",),
+    ThermostatMode.AUTO_CHANGE_OVER: ("setpoint_auto_changeover",),
+    ThermostatMode.HEATING_ECON: ("setpoint_eco_heating",),
+    ThermostatMode.COOLING_ECON: ("setpoint_eco_cooling",),
+    ThermostatMode.AWAY: ("setpoint_away_heating", "setpoint_away_cooling"),
+    ThermostatMode.FULL_POWER: ("setpoint_full_power",),
 }
 
+
 # strings, OZW and/or qt-ozw does not send numeric values
+# https://github.com/OpenZWave/open-zwave/blob/master/cpp/src/command_classes/ThermostatOperatingState.cpp
 HVAC_CURRENT_MAPPINGS = {
     "idle": CURRENT_HVAC_IDLE,
```

Comment:
Would be good with constants/enums in the library for the left part of this dict.